Sources/_StringProcessing/RegexDSL/DSL.swift

```diff
@@ -13,13 +13,22 @@ import _MatchingEngine
 
 // MARK: - Primitives
 
+private func concat(
```

Comment:
Looks unused, and not sure if it's any better than `.concatenation(...)` at call sites.